### PR TITLE
Add option to suppress warnings when querying Redis

### PR DESCRIPTION
**Pull request by TACappleman**
In some circumstances calls are made to query the Redis database without checking if a given key (or key/field pair) exists, and with code to handle the case where it does not. For example many show commands query all possible relevant data, without checking if it exists. In these cases it would be useful to be able to prevent warning logs being written for every missing field, and so we add that option in this PR.

### CODE REVIEW

**Comment by TACappleman** [comment]
@qiluo-msft could you take a look at this PR please (looks like you've been the most recent approver in this PR)?

This is a very minor utility change that will be useful in preventing unnecessary warning logs in certain scenarios - e.g. when show commands query Redis for many fields without checking if they're present, and are able to handle the case where they're not present.

File: src/swsssdk/interface.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def keys(self, db_name, pattern='*'):
             return keys
 
     @blockable
```
Comment (qiluo-msft) [inline src/swsssdk/interface.py:282]: >blockable [](http://example.com/codeflow?start=5&length=9)

We are deprecate the python implementation and in favor of swss-common (SWIG powered). Would you like to change in that repo at the same time or only in that repo?

File: src/swsssdk/interface.py [hunk 2]
```diff
@@ -280,7 +280,7 @@ def keys(self, db_name, pattern='*'):
...
-    def get(self, db_name, _hash, key):
+    def get(self, db_name, _hash, key, log_warning=True):
```
Comment (qiluo-msft) [inline src/swsssdk/interface.py:283]: >log_warning [](http://example.com/codeflow?start=39&length=11)

name it as `verbose` ?

File: src/swsssdk/interface.py (hunk 1 above)
Comment (TACappleman) [reply src/swsssdk/interface.py:282]: Made another PR for that here: https://github.com/Azure/sonic-swss-common/pull/566

File: src/swsssdk/interface.py (hunk 2 above)
Comment (TACappleman) [reply src/swsssdk/interface.py]: Done

**Comment by qiluo-msft** [comment]
>                 if blocking:

I have second thought about the solution. Since the functions are throwing exceptions with the message, it is the caller's responsibility to handle and log

0. We should remove the logging statements from the basic functions `keys`/`get`/`get_all`/etc, if it is followed by a throw.
1. We may add the logging statements into the `blocking` branch in the function `wrapped`. Then no behavior change in this condition.
2. For non blocking branch, we swallow the `UnavailableDataError` and `return None`. The caller should get well understanding what happening, and even the caller does not handle well, the further exception on None object should be clear. So I think we are okay with the behavior change.

What do you think?




---
Refers to: src/swsssdk/interface.py:42 in b8838a8. [](commit_id = b8838a8290a7bc0bdebbc455196f7faed3b7d819, deletion_comment = False)

**Comment by TACappleman** [comment]
> > ```
> >             if blocking:
> > ```
> 
> I have second thought about the solution. Since the functions are throwing exceptions with the message, it is the caller's responsibility to handle and log
> 
> 0. We should remove the logging statements from the basic functions `keys`/`get`/`get_all`/etc, if it is followed by a throw.
> 1. We may add the logging statements into the `blocking` branch in the function `wrapped`. Then no behavior change in this condition.
> 2. For non blocking branch, we swallow the `UnavailableDataError` and `return None`. The caller should get well understanding what happening, and even the caller does not handle well, the further exception on None object should be clear. So I think we are okay with the behavior change.
> 
> What do you think?
> 
> Refers to: src/swsssdk/interface.py:42 in [b8838a8](https://github.com/Azure/sonic-py-swsssdk/commit/b8838a8290a7bc0bdebbc455196f7faed3b7d819). [](commit_id = [b8838a8](https://github.com/Azure/sonic-py-swsssdk/commit/b8838a8290a7bc0bdebbc455196f7faed3b7d819), deletion_comment = False)

Works for me. I've updated this and the swss-common equivalent: https://github.com/Azure/sonic-swss-common/pull/566

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**